### ISSUE TITLE
allow bypass of embedded drv.nix

### ISSUE THREAD

**Issue by johanot** [wharfix/wharfix#7]
drv.nix is the Nix expression that builds the servable directory that wharfix consumes when serving manifest- and blob requests. 

Input to drv.nix is an expression in the form that dockerTools.buildLayeredImage wants it and output is a v2 manifest along with some decompressed tarballs.

We want this intermediate Wharfix build expression to be injectable from _outside_ wharfix. The embedded version of the expression _inside_ the wharfix code base is useful for simple usecases, like: https://github.com/wharfix/examples . However, when used with DBC deployments, we want full control over the entire build expression from inside our internal deployments repo.

TLDR; Wharfix should be able to: `nix-build <indexFile> -A <image-name>` (if configured to do so), rather than the current behavior: `nix-build ./drv.nix --arg indexFile <index-file> -A <image-name>`.
